radio/radio_model.py

```diff
@@ -199,6 +210,9 @@ def forward_intermediates(
             intermediates_only=intermediates_only,
             aggregation=aggregation,
         )
+
+        outputs = [(summary, self.feature_normalizer(features)) for summary, features in outputs]
+
```

Comment:
I modified this piece of code on my end:

```
        intermediates = self.model.forward_intermediates(
            x,
            indices=indices,
            return_prefix_tokens=return_prefix_tokens,
            norm=norm,
            stop_early=stop_early,
            output_fmt=output_fmt,
            intermediates_only=intermediates_only,
            aggregation=aggregation,
        )
 
        if not intermediates_only:
            final, intermediates = intermediates

        if return_prefix_tokens:
            intermediates = [(summary, self.feature_normalizer(features)) for summary, features in intermediates]
            radio_outputs = [RadioOutput(summary, features) for (summary, features) in intermediates]
        else:
            intermediates = [self.feature_normalizer(features) for features in intermediates]
            radio_outputs = [RadioOutput(None, features) for features in intermediates]

        if intermediates_only:
            return radio_outputs
        else:
            return self.feature_normalizer(final), radio_outputs
```

Reply:
What about this?
```
        if return_prefix_tokens:
            radio_outputs = [
                RadioOutput(summary, self.feature_normalizer(features)) 
                for summary, features in intermediates
            ]
            radio_outputs = [RadioOutput(summary, features) for (summary, features) in intermediates]
        else:
            radio_outputs = [
                self.feature_normalizer(features) 
                for features in intermediates
            ]

        if intermediates_only:
            return radio_outputs
        else:
            return self.feature_normalizer(final), radio_outputs
```

What is `final` in this case? Is it itself a tuple? Or is it a tensor with all of the cls tokens, registers, and patches concatenated? If the latter, then I suppose we need to extract the cls tokens and patches, and normalize the patches.

Reply:
Maybe now the docstring should specify that a list of `RadioOutput` is return if `return_prefix_tokens==True` otherwise it will be a list of feature tensors.